crates/red_knot_test/src/matcher.rs

```diff
@@ -67,6 +75,29 @@ where
     let mut current_assertions = line_assertions.next();
     let mut current_diagnostics = line_diagnostics.next();
 
+    // If this test has requested a snapshot and there are zero
+    // inline assertions in the source code, then we skip checking
+    // the assertions for this file. The logic here is that some
+    // tests might only be interested in diagnostic output, and it
+    // is otherwise laborious to annotate all inline errors. And
+    // particularly so when these are covered by snapshots.
+    //
+    // However, one might still want to have inline assertions with a
+    // snapshot, perhaps when a test doubles as one for type checking
+    // and for diagnostic output.
+    //
+    // I took this route since it seemed like the easiest thing to
+    // grok, but there are potentially other configurations. For
+    // example, making inline assertions and snapshots completely
+    // mutually exclusive (if you use one you can't do the other).
+    if matches!(
+        mode,
+        MatchFileMode::SkipUnmatchedDiagnosticsWithoutAssertions
+    ) && current_assertions.is_none()
```

Comment:
nit: add a `MatchFileMode::should_skip_unmatched_diagnostics_without_assertions()` method?

Reply:
I could go either way here to be honest. But I ended up deleting this.